services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,55 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  if(thermalMgrQueueHandle == NULL){
+    return ERR_CODE_INVALID_STATE;
+  }
+
+  if(event == NULL){
+    return ERR_CODE_INVALID_ARG;
+  }
 
+  if(xQueueSend(thermalMgrQueueHandle, event, 0) == errQUEUE_FULL){
+    return ERR_CODE_QUEUE_FULL;
+  }
+  
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  float temp;
+  readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
+
+  if(temp > LM75BD_DEFAULT_HYST_THRESH){
+    overTemperatureDetected();
+  }
+  else if(temp <= LM75BD_DEFAULT_HYST_THRESH){
+    safeOperatingConditions();
+  }
+
```

Comment:
As this is being called in it should be short, send an event to the queue indicating that it is an overtemperature event